pkg/apis/build/v1alpha2/build_conversion_test.go

```diff
@@ -113,48 +188,72 @@ func testBuildConversion(t *testing.T, when spec.G, it spec.S) {
 				SubPath: "my-sub-path",
 			}
 
-			v1alpha1Build := &v1alpha1.Build{}
-			err := build.DeepCopy().ConvertTo(context.TODO(), v1alpha1Build)
+			testV1alpha1Build := &v1alpha1.Build{}
+			err := v1alpha2Build.ConvertTo(context.TODO(), testV1alpha1Build)
 			require.NoError(t, err)
+			require.Equal(t, testV1alpha1Build, v1alpha1Build)
 
-			convertedBackBuild := &Build{}
-			err = convertedBackBuild.ConvertFrom(context.TODO(), v1alpha1Build)
+			testV1alpha2Build := &Build{}
+			err = testV1alpha2Build.ConvertFrom(context.TODO(), v1alpha1Build)
 			require.NoError(t, err)
-
-			require.Equal(t, build, convertedBackBuild)
+			require.Equal(t, testV1alpha2Build, v1alpha2Build)
 		})
 
-		it("does not have any data loss with blob source", func() {
-			build.Spec.Source = corev1alpha1.SourceConfig{
+		it("converts with blob source", func() {
+			v1alpha2Build.Spec.Source = corev1alpha1.SourceConfig{
+				Blob: &corev1alpha1.Blob{
+					URL: "my-blob.com",
+				},
+				SubPath: "my-sub-path",
+			}
+			v1alpha1Build.Spec.Source = corev1alpha1.SourceConfig{
 				Blob: &corev1alpha1.Blob{
 					URL: "my-blob.com",
 				},
 				SubPath: "my-sub-path",
 			}
 
-			v1alpha1Build := &v1alpha1.Build{}
-			err := build.DeepCopy().ConvertTo(context.TODO(), v1alpha1Build)
+			testV1alpha1Build := &v1alpha1.Build{}
+			err := v1alpha2Build.ConvertTo(context.TODO(), testV1alpha1Build)
 			require.NoError(t, err)
+			require.Equal(t, testV1alpha1Build, v1alpha1Build)
 
-			convertedBackBuild := &Build{}
-			err = convertedBackBuild.ConvertFrom(context.TODO(), v1alpha1Build)
+			testV1alpha2Build := &Build{}
+			err = testV1alpha2Build.ConvertFrom(context.TODO(), v1alpha1Build)
 			require.NoError(t, err)
+			require.Equal(t, testV1alpha2Build, v1alpha2Build)
+		})
 
-			require.Equal(t, build, convertedBackBuild)
+		it("can't convert new fields to v1alpha1", func() {
```

Comment:
What is this actually testing? That those fields don't cause it to error? Has the test ever failed?

Reply:
yea it doesn't really do anything. I just wanted to acknowledge the fields that don't convert to v1alpha1 and are dropped. I don't think this really tests anything.